x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json

```diff
@@ -0,0 +1,974 @@
+    },
+    {
+      "set" : {
+        "field": "client.user.name",
+        "value": "{{checkpoint.administrator}}",
+        "if": "ctx.checkpoint?.administrator != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "source.user.name",
+        "value": "{{checkpoint.administrator}}",
+        "if": "ctx.checkpoint?.administrator != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "source.packets",
+        "value": "{{ctx.checkpoint.client_outbound_packets}}",
+        "if": "ctx.checkpoint?.client_outbound_packets != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "client.packets",
+        "value": "{{ctx.checkpoint.client_outbound_packets}}",
+        "if": "ctx.checkpoint?.client_outbound_packets != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "destination.packets",
+        "value": "{{checkpoint.server_outbound_packets}}",
+        "if": "ctx.checkpoint?.server_outbound_packets != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "server.packets",
+        "value": "{{checkpoint.server_outbound_packets}}",
+        "if": "ctx.checkpoint?.server_outbound_packets != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "client.bytes",
+        "value": "{{checkpoint.client_outbound_bytes}}",
+        "if": "ctx.checkpoint?.client_outbound_bytes != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "source.bytes",
+        "value": "{{checkpoint.client_outbound_bytes}}",
+        "if": "ctx.checkpoint?.client_outbound_bytes != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "destination.bytes",
+        "value": "{{checkpoint.server_outbound_bytes}}",
+        "if": "ctx.checkpoint?.server_outbound_bytes != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "server.bytes",
+        "value": "{{checkpoint.server_outbound_bytes}}",
+        "if": "ctx.checkpoint?.server_outbound_bytes != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "destination.port",
+        "value": "{{checkpoint.service}}",
+        "if": "ctx.checkpoint?.service != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "server.port",
+        "value": "{{checkpoint.service}}",
+        "if": "ctx.checkpoint?.service != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "client.port",
+        "value": "{{checkpoint.s_port}}",
+        "if": "ctx.checkpoint?.s_port != null"
+      }
+    },
+    {
+      "set" : {
+        "field": "source.port",
+        "value": "{{checkpoint.s_port}}",
+        "if": "ctx.checkpoint?.s_port != null"
+      }
+    },
+    {
+      "set": {
+        "field": "event.category",
+        "value": "authentication",
+        "if": "ctx.checkpoint?.operation == 'Log In'"
+      }
+    },
+    {
+      "set" : {
+        "field": "event.kind",
+        "value": "alert",
+        "if": "['Prevent', 'Detect', 'Quarantine'].contains(ctx.checkpoint?.rule_action)"
```

Comment:
can we use Prevent, Detect or Quarantine to determine if "malware" or "intrusion_detection" should be added to event.category?

Reply:
Added malware if malware_action was not null + intrusion_detection if the Checkpoint.rule_action is prevent or detect (as those are the only used by IPS)